### PR TITLE
Update URL Checker workflow

### PR DESCRIPTION
**Pull request by arkid15r**
This Pull Request relates to issue #1990 
This PR **does not** address the deprecated action update.

Update URL Checker workflow
  - merge 2 URL checker workflows (push, pr triggers)
  - limit runs to 5.0/** paths
  - bump actions/checkout from v3 to v4

### CODE REVIEW

File: .github/workflows/url-checker.yml [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
-          fetch-depth: 1
-      - name: link-check
+        uses: actions/checkout@v4
+
+      - name: Set options
+        id: options
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "check-modified-files-only=yes" >> $GITHUB_OUTPUT
+            echo "use-quiet-mode=yes" >> $GITHUB_OUTPUT
```
Comment (tghosth) [inline .github/workflows/url-checker.yml:26]: Would it be clearer to put this outside the if statement if the answer is the same either way or even move it back to being hard coded?

File: .github/workflows/url-checker.yml [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
+      - name: Check URLs
         uses: gaurav-nelson/github-action-markdown-link-check@v1
         with:
-          use-quiet-mode: 'yes'
-          use-verbose-mode: 'no'
-          config-file: '.github/workflows/config/url-checker-config.json'
-          folder-path: './5.0/'
+          check-modified-files-only: ${{ steps.options.outputs.check-modified-files-only }}
+          config-file: .github/workflows/config/url-checker-config.json
+          folder-path: '5.0'
```
Comment (tghosth) [inline .github/workflows/url-checker.yml:39]: Does this change to the folder path make sure it only looks at links within 5.0?

File: .github/workflows/url-checker.yml [hunk 3]
```diff
@@ -1,23 +1,41 @@
...
+  workflow_dispatch:
 
 jobs:
   markdown-link-check:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 1
```
Comment (tghosth) [inline .github/workflows/url-checker.yml:16]: Do we no longer need fetch depth?

**Review by tghosth — CHANGES_REQUESTED** [review]

File: .github/workflows/url-checker.yml (hunk 3 above)
Comment (arkid15r) [reply .github/workflows/url-checker.yml:16]: That's correct, `1` is the [default value](https://github.com/actions/checkout?tab=readme-ov-file#checkout-v4).

File: .github/workflows/url-checker.yml (hunk 1 above)
Comment (arkid15r) [reply .github/workflows/url-checker.yml:26]: Good point, done.

File: .github/workflows/url-checker.yml (hunk 2 above)
Comment (arkid15r) [reply .github/workflows/url-checker.yml:39]: I've modified the `on` part with `paths` options to make sure it runs only when '5.0/**' files are affected.

The triggers also include `workflow_dispatch` and `folder-path: '5.0'` should take care of those cases.

**Review by arkid15r — COMMENTED** [review]
@tghosth thanks for reviewing:

**Comment by tghosth** [comment]
Hi @arkid15r, thanks for your responses but I don't see that you actually made the changes to the PR?

**Comment by arkid15r** [comment]
> Hi @arkid15r, thanks for your responses but I don't see that you actually made the changes to the branch in the PR?

Sorry for confusion. I forgot to update the PR yesterday.

**Review by tghosth — APPROVED** [review]

**Pull request merged**